### PR TITLE
Introduce strict TypeScript

### LINKED ISSUE #763: Broken on TypeScript
Hello, I am trying to use this package with TypeScript, but starting my application fails. I posted the log on Pastebin, because it's too long to include here: https://pastebin.com/Dn7VT1gG

To reproduce this, simply trying to run any Node project using this TypeScript with this package should be enough.

### PR DESCRIPTION
**Pull request by Borewit**
Resolves #763

### CODE REVIEW

File: lib/musicbrainz-api.ts [hunk 1]
```diff
@@ -544,7 +546,7 @@ export class MusicBrainzApi {
     const path = `ws/2/${entity}/`;
     // Get digest challenge
 
-    let digest: string = null;
+    let digest: string | null = null;
```
Comment (Haschikeks) [inline lib/musicbrainz-api.ts:549]: This could be `string | undefined` to make line 558 easier to write.

File: lib/digest-auth.ts [hunk 2]
```diff
@@ -86,11 +86,11 @@ export class DigestAuth {
 
     const parts: string[] = [];
     for (const k in authValues) {
-      if (authValues[k]) {
+      if ((authValues as any)[k]) {
```
Comment (Haschikeks) [inline lib/digest-auth.ts:89]: `as any` does not look nice. Maybe we can use `let authValue = authValues[k as keyof typeof authValues];` and then use the now typed `authValue` in the next steps.

File: lib/musicbrainz.types.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ export interface IArtist extends IEntity {
   disambiguation: string;
   'sort-name': string;
   'type-id'?: string;
-  'gender-id'?;
+  'gender-id'?: any
```
Comment (Haschikeks) [inline lib/musicbrainz.types.ts:42]: This is probably a `string` as well?

File: lib/digest-auth.ts (hunk 2 above)
Comment (Borewit) [reply lib/digest-auth.ts:89]: Good suggestion, but I will follow this guidance https://imfaber.me/typescript-how-to-iterate-over-object-properties/#use-objectentries

File: lib/musicbrainz.types.ts (hunk 3 above)
Comment (Borewit) [reply lib/musicbrainz.types.ts:42]: Let's assume so indeed

**Pull request merged**